code/game/objects/items/devices/scanners.dm

```diff
@@ -300,6 +301,12 @@ MASS SPECTROMETER
 		else
 			user << "<span class='info'>Plasma: [round(plasma_concentration*100, 0.01)] %</span>"
 
+		if(bz_concentration > 0.005)
+			user << "<span class='alert'>Hallucinogen: [round(plasma_concentration*100, 0.01)] %</span>"
+		else
+			user << "<span class='info'>Hallucinogen: [round(plasma_concentration*100, 0.01)] %</span>"
+

```

Comment:
The fact that this whole proc uses if column is horrible
For loop over gases, add "dangerous concentration level" var to gases, check the real concentration of the gas compared to the gases var and output the name of the gas